### ISSUE TITLE
id:min and id:max is not working on v3 catalog/prodcuts api ?

### ISSUE THREAD

**Issue by rakeshdobariya** [bigcommerce/api#113]
have 65K products in my catalog and one of my product id 2121. I have requirement to download full catalog and then onwards new products. I am trying to below call to get all products

https://api.bigcommerce.com/stores/{{ hash }}/v3/catalog/products?id:min=1

### Expected behavior
Giving me first page or products

### Actual behavior

Nothing is returned

### Steps to reproduce behavior

https://api.bigcommerce.com/stores/{{ hash }}/v3/catalog/products?id:min=1

Replace store hash.

**Comment by tatiana-perry** [comment]
Hi, right now you are trying to pull all your products at once using a min ID of 1. You can only return up to 250 products per page. You will need to write a loop that filters through the pages.

**Comment by rakeshdobariya** [comment]
Yes, Pulling all product is my FINAL goal. Expected behaviour is I should get 250 records. But problem is I am not getting any record at all. I have all logic in place, but first of all API should return first 250 record when I request with parameter id:min=1. But it is not returning any data. That's why I reported here.

**Comment by rakeshdobariya** [comment]
Attached is screen shot of post man
<img width="1395" alt="screen shot 2018-10-13 at 11 11 06 pm" src="https://user-images.githubusercontent.com/8028922/46908225-75bbd980-cf3d-11e8-84aa-5fad4fd99725.png">

**Comment by tatiana-perry** [comment]
Remove the min id of 1. Then run the request.

**Comment by rakeshdobariya** [comment]
Hi,

Thanks for reply. I think we are not getting to the point here. May be I am not explaining problem in deep. 
Just my back-ground : I am technical partner and having 4 year experience working in BigCommerce APIs.

So. I have requirement to download full catalog for one of my client and do insert incremental products that my client add. 

Simple algo. Initially get MAX id from database, if you do not found anything then MAX = 1

Get first page using catalog product api, where id:min = value from my database.  Here if I give id:min=1, its not returning any data, practically it should return default 50 records.

As per suggestion by @tatiana-perry  I removed id:min , it gives me 50 records. Saved in database all good so far. 

Now I need to get next 50 records. So I will fetch max from database. in my case it is 57. So my next call should be 

https://api.bigcommerce.com/stores/{{hasg}}/v3/catalog/products?id:min=57 

practically it should give next 50 records. ? Please correct me if I am wrong. But In this case it is giving me only 1 record with 57.

So, either its my misunderstanding regarding attribute:min  or it is a bug in v3. 

I do understand there are alternate ways to get full catalog, like using page reference in metadata. But my target here is not to get solution. My target here is to showcase bug or correct my misunderstanding.

**Comment by tatiana-perry** [comment]
I was able to confirm there is a bug where you can not use the `product_id` as a filter for products. You can use it to return the exact `product_id` you are trying to retrieve, but not use it as a filter query for min and max.

**Comment by rakeshdobariya** [comment]
@tatiana-perry 

This is serious bug, there are no perfect alternative to download products. Can you please respond ?

**Comment by tatiana-perry** [comment]
Hi, the other alternative to download products is to loop through each page. Once a fix is deployed we will make an announcement on our [changelog](https://developer.bigcommerce.com/changelog/). Is there a reason why looping through the pages is not a good workaround for you?

**Comment by rakeshdobariya** [comment]
Yes, There is reason I can not rely on page loop. For code performance, we have written code that will work for new download as well as continues download. 

So say, when we are downloading products very first time, I will query database which is latest product I have, no result found then program will query API with min Id, else it will query API Id.

Can you suggest alternate way to get  this ?

**Comment by karen-white** [comment]
@rakeshdobariya Hey Rakesh, we've opened an engineering ticket to address this. I don't have an ETA for a fix just yet, but I'll keep you posted. Thank you for reporting the bug and sharing the details of your use case to help frame the priority.

I did want to share that we're working on adding a filter parameter for date_created, so in the near future that could be another way to accomplish downloading newly added products. In the meantime, our team will prioritize this as a bug fix.

**Comment by rakeshdobariya** [comment]
That really good news 


Sent from Yahoo Mail for iPhone


On Thursday, January 10, 2019, 5:58 AM, Karen White <notifications@github.com> wrote:


Hi @rakeshdobariya , I wanted to give you an update on this! We've begun rolling out a fix. It hasn't yet reached all production tiers, but it should be platform-wide within the next few days.
